### PR TITLE
COL-744, asset-library-search gets new filter: by Section

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/COL-744

List of distinct section names can be extracted from the list `users` already available in angular directive. SQL behind search now has `array_position(u.canvas_course_sections, :section)`

image

Initial PR #390 was closed due to no test coverage.

### CODE REVIEW

File: node_modules/col-assets/tests/test-assets.js [hunk 1]
```diff
@@ -40,6 +41,64 @@ var AssetsTestUtil = require('./util');
...
+        'canvas_full_name': canvasUser.fullName,
+        'canvas_course_sections': sections
+      };
+      UsersAPI.getOrCreateUser(canvasUser.id, course, profile, function(err, user) {
+        assert.ok(!err);
+
+        callback(user);
+      });
+    });
+  };
```
Comment (johncrossman) [inline node_modules/col-assets/tests/test-assets.js]: Like `test-poller`, we have direct interaction with db. As far as I know, this is the simplest way to assign sections to the user before verifying search.

Comment (paulkerschen) [inline node_modules/col-assets/tests/test-assets.js]: I agree, @johncrossman, straight into the db is the way to go.

**Comment by johncrossman** [comment]
@paulkerschen Postgres 9.5 makes Travis a happy boy. I'll up bump AWS if and when this is merged.

**Comment by paulkerschen** [comment]
THANK you, @johncrossman.

**Pull request merged**